Develop blaine

lindonbej opened the pull request.

adamqure commented:
I'm looking through this and in general, it is looking good. I've connected the main activity to your fragment and fixed R.

Great points:
- The layout looks good and complete. A few UI tweaks and it'll be perfect
- Good management in the view
- Thank you for keeping the Presenter connected to the interfaces in the Contract

 I do have a few comments though:

First, to answer your question about not connecting to a presenter. We need the presenter to let the model send a response to the View, and the view to send a request to the model.

Second, there are some endcase things to check. Specifically, if the user doesn't enter any information into the EditTexts, the strings are null, therefore, a comparator is going to throw a null pointer. You need to make sure the register button doesn't work until the use has entered any information.

Third, I'm trying to see where you want me to send the user back to the main activity... 

Lastly, in the layout, I don't see the hints in the EditTexts, the EditTexts aren't the same width, and you can't really see the register greeting. I'd suggest grabbing the font from my branch and using a maroon color, unless you can figure out a better color (I have one defined in my Colors.xml that I've been using).

lindonbej committed:
Merge branch 'develop' into develop-blaine

lindonbej committed:
Fixed null pointer exception
Changed register methods to fit presenter model
RegisterFragment exits on success
Probably broke root gradle file (but it was already broken)

lindonbej committed:
Merge remote-tracking branch 'origin/develop-blaine' into develop-blaine

# Conflicts:
#	app/src/main/java/com/example/rholbrook/tickettoride/main/MainActivityModel.java
#	app/src/main/java/com/example/rholbrook/tickettoride/main/MainActivityPresenter.java
#	app/src/main/java/com/example/rholbrook/tickettoride/serverconnection/ClientFacade.java
#	app/src/main/java/com/example/rholbrook/tickettoride/serverconnection/ServerProxy.java

lindonbej committed:
Fixed Register bugs -
Made button disabled initially
Fixed editText widths
Fixed greeting font and color
Changed Model to update Authentication username upon registration success

adamqure reviewed (changes requested):
You know the changes needed

lindonbej committed:
Made model/presenter into Obervable/Observer

lindonbej committed:
Tweaked register edittexts

lindonbej committed:
Merge branch 'develop' into develop-blaine

lindonbej committed:
Removed Kotlin from Gradle files

lindonbej committed:
Just making sure I pushed everything

adamqure committed:
Merge branch 'develop' into develop-blaine

adamqure commented:
Oh the merge conflicts!!! Just spend 30 minutes on all the conflicts.... Oh well. Thanks for the great work!
